scripts/prices.js

```diff
@@ -1,88 +1,136 @@
 import { getEventEmitter } from './event_bus.js';
-import { isEmpty } from './misc.js';
-import { cMarket, fillCurrencySelect } from './settings.js';
+import { cOracle, fillCurrencySelect, strCurrency } from './settings.js';
 
 /**
- * CoinGecko's endpoint for PIVX data, optimised for least bandwidth
- * - No localisation, tickers, community data, developer data or sparklines
+ * @typedef {Object} Currency
+ * @property {string} currency - The type of currency
+ * @property {number} value - The value of the currency
+ * @property {number} last_updated - The timestamp when this value was last updated
  */
-export const COINGECKO_ENDPOINT =
-    'https://api.coingecko.com/api/v3/coins/pivx?localization=false&tickers=false&market_data=true&community_data=false&developer_data=false&sparkline=false';
 
 /**
- * The generic market data source template, used to build site-specific classes
+ * Oracle's primary instance.
+ *
+ * @todo Allow an array of Oracle instances for better privacy and decentralisation
  */
-export class MarketSource {
-    /** The storage object for raw market data */
-    cData = {};
+export const ORACLE_BASE = 'https://pivxla.bz/oracle/api/v1';
 
-    /** The name of the market source */
-    strName = '';
-
-    /** The customised API endpoint of the market source */
-    strEndpoint = '';
+/**
+ * An Oracle instance
+ */
+export class Oracle {
+    /**
+     * The currencies cache list
+     * @type {Currency[]} Array to store currency objects
+     */
+    arrCurrencies = [];
```

Comment:
Instead of using an array use an unordered map, I suggest:
map `strCurrency` -> `Currency `
It would make the code much cleaner (and more optimized)

Reply:
Done, much nicer cache setting/getting now for sure.